app/javascript/components/util/DateInput.js

```diff
@@ -3,34 +3,54 @@ import PropTypes from 'prop-types';
 import { OverlayTrigger, Tooltip } from 'react-bootstrap';
 import DatePicker from 'react-datepicker';
 import MaskedInput from 'react-text-mask';
-
 import moment from 'moment';
 
 class DateInput extends React.Component {
   constructor(props) {
     super(props);
     this.datePickerRef = React.createRef();
-    this.handleRawChange = this.handleRawChange.bind(this);
-    this.handleDateChange = this.handleDateChange.bind(this);
-    this.clearDate = this.clearDate.bind(this);
+    this.state = {
+      lastValidDate: props.date || null,
+    };
   }
 
-  handleDateChange(date) {
-    this.props.onChange(date && moment(date).format('YYYY-MM-DD'));
+  handleDateChange = date => {
+    const momentDate = date && moment(date).format('YYYY-MM-DD');
+    this.props.onChange(momentDate);
     this.datePickerRef.current.setOpen(false);
-  }
+    if (this.validDate(momentDate)) {
+      this.setState({ lastValidDate: momentDate });
+    }
+  };
 
-  handleRawChange(event) {
+  handleRawChange = event => {
     if (event.target.value) {
       this.datePickerRef.current.setOpen(true);
     }
+  };
+
+  handleOnBlur = () => {
```

Comment:
Could you add a method comment to explain what the `onBlur` event handler is to those who may not be familiar?

Reply:
done here: https://github.com/SaraAlert/SaraAlert/pull/734/commits/5f20fac4fecdfd502e965f953c798387862c84e2